pathway-analysis/02-gene_set_enrichment_analysis.Rmd

```diff
@@ -111,77 +107,143 @@ dge_results_df <- readr::read_tsv(dge_results_file)
 head(dge_results_df)
 ```
 
-We can take the same steps we did earlier to convert these Ensembl gene IDs to Entrez IDs. 
-We have to change the `column` argument to `mapIds()` (and the downstream steps).
+Since this data frame of DGE results includes gene symbols, we do not need to perform any kind of gene identifier conversion.
+We do, however, need to check for duplicate gene symbols. 
+We can accomplish this with `duplicated()`, which returns a logical vector (e.g., `TRUE` or `FALSE`).
+The function `any()` will return `TRUE` if **any** value it is given `TRUE`.
+
+```{r any_duplicated, live = TRUE}
+any(duplicated(dge_results_df$gene_symbol))
+```
```

Comment:
Later you repeat this with `sum()` so why not just do that here and save a bit of duplication?